From 27470b650c59ab70580a93c5a69f54e7761764c6 Mon Sep 17 00:00:00 2001
From: anonym <anonym@riseup.net>
Date: Tue, 3 Nov 2020 12:13:10 +0100
Subject: [PATCH] Don't reset encryption choice when disabling encryption
 controls.

Without this, the mail.identity.default.encryptionpolicy pref does not
work: adding a key to an account always set its encryptionpolicy to 0
no matter the desired default.

While we're at it, also kill two lines of code before a call to
enableEncryptionControl() that exactly replicates it.
---
 comm/mail/extensions/am-e2e/am-e2e.js | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/comm/mail/extensions/am-e2e/am-e2e.js b/comm/mail/extensions/am-e2e/am-e2e.js
index d70bd5d415b..d01d05c9b96 100644
--- a/comm/mail/extensions/am-e2e/am-e2e.js
+++ b/comm/mail/extensions/am-e2e/am-e2e.js
@@ -442,9 +442,6 @@ function smimeSelectCert(smime_cert) {
 function enableEncryptionControls(do_enable) {
   gRequireEncrypt.disabled = !do_enable;
   gDoNotEncrypt.disabled = !do_enable;
-  if (!do_enable) {
-    gEncryptionChoices.value = 0;
-  }
 }
 
 function enableSigningControls(do_enable) {
-- 
2.31.0

